### PR TITLE
Implemented notification service for delivering daily emails

### LINKED ISSUE #117: Send daily emails to predefined list of addresses
We need to create a new entity that will hold information about the public entities we are sending emails to. The entity will have this information:
- id
- county
- name of public entity
- email address of public entity

At the end of the day we need to send information to these state entities about the answers posted by users in the platform.
The emails will be sent everyday at 20:00 - we need to enhance the Scheduling service for this task.
We need to generate a new Notification service that will select all the user answers for the interval: 20:00 previous day, 20:00 today.
We need to use the excel service to generate excel files based on the response of users.
The excel file will be sent as attachment in the email to the public entities.

### PR DESCRIPTION
**Pull request by bogdanbujdea**
### Requirements for making a pull request

Thank you for contributing to our project! 

Please fill out the template below to help the project maintainers review it as fast as possible and include your contribution to the project.

### What does it fix?

Closes #117 

Please mention the main changes this PR brings.

- I implemented the ResultNotificationsDispatch service which sends emails daily at 8PM
- I had to move the services for creating Excel files inside the Common project
- I implemented attachments for both the SendGrid sender and the SMTP sender
- The list of emails for each county is now retrieved from appsettings.json instead of the database

### What's missing:
- handle form dates correctly, either using DateTimeOffset or storing UTC instead of local time
- use an HTML for the content of the email
- find out if SendGrid or the SMTP service have any rate limits because now we send all the emails in parallel and this may cause issues

### How has it been tested?
Local machine, created forms with the web app, triggered the JobScheduler and used Google's SMTP server to send email and test that the attachments work correctly

Please describe the tests that you ran to verify your changes.

### CODE REVIEW

File: backend/src/StamAcasa.JobScheduler/Program.cs [hunk 1]
```diff
@@ -44,18 +48,38 @@ private static void Main(string[] args)
...
+
                     services.TryAddScheduledJob<SendResultsNotificationJob>(s =>
                     {
                         s.TryAddTransient<ResultNotificationsDispatch>();
+                        s.TryAddTransient<IFormService, FormService>();
+                        s.TryAddTransient<IAnswersExcelExporter, AnswersExcelExporter>();
+                        s.TryAddTransient<IExcelDocumentService, ExcelDocumentService>();
+                        s.AddSingleton<IEmailSender>(ctx =>
+                            new SmtpSender(
+                                new SmtpOptions
```
Comment (trupci) [inline backend/src/StamAcasa.JobScheduler/Program.cs]: But should we be using the IEmailSender directly or the `IQueueService` ? Like we do  [here](https://github.com/code4romania/stam-acasa/blob/4e57bbbecc372776c3105d1db8642776473e3985/backend/src/StamAcasa.IdentityServer/Areas/Identity/Pages/Account/Register.cshtml.cs#L90)

Comment (bogdanbujdea) [reply backend/src/StamAcasa.JobScheduler/Program.cs]: You're right, I didn't know about the IQueueService, so I thought we will either use the SendGrid sender or the SMTP implementation. I'll push another commit and replace it with IQueueService.

File: backend/src/StamAcasa.Common/Services/Emailing/SmtpSender.cs [hunk 2]
```diff
@@ -26,13 +27,33 @@ public async Task SendAsync(Email email, CancellationToken cancellationToken)
...
                 {
                     Sender = new MailboxAddress(email.FromName, email.FromEmail),
                     Subject = email.Subject,
-                    Body = new TextPart(TextFormat.Html) { Text = email.Content },
+                    Body = body,
                     To = { new MailboxAddress(email.To) }
                 };
+                if (email.Attachment != null)
+                {
+                    var attachment = new MimePart("image", "gif")
```
Comment (RaduCStefanescu) [inline backend/src/StamAcasa.Common/Services/Emailing/SmtpSender.cs]: Is this the correct Mime type?
Just asking, looks wierd :)

File: backend/src/StamAcasa.JobScheduler/Program.cs (hunk 1 above)
Comment (RaduCStefanescu) [reply backend/src/StamAcasa.JobScheduler/Program.cs]: Yup, please switch to the IQueueService here. Thank you!

File: backend/src/StamAcasa.JobScheduler/appsettings.json [hunk 3]
```diff
@@ -23,6 +23,19 @@
     "StamAcasa.JobScheduler.Jobs.SendAssessmentReminderJob": "0 9,18 * * *",
 
     // Job that sends result notifications and runs daily at 07:00.
-    "StamAcasa.JobScheduler.Jobs.SendResultsNotificationJob": "0 7 * * *"
+    "StamAcasa.JobScheduler.Jobs.SendResultsNotificationJob": "0 20 * * *"
+  },
+  "CountyEmailDistribution": {
+    "sender": "code4contact@code4.ro",
+    "templateFile": "whatever.html",
```
Comment (RaduCStefanescu) [inline backend/src/StamAcasa.JobScheduler/appsettings.json]: I don't think we need the template here.
We will use templates based on the enumeration we are using in the Email Service.

File: backend/src/StamAcasa.Common/Services/Emailing/SmtpSender.cs (hunk 2 above)
Comment (bogdanbujdea) [reply backend/src/StamAcasa.Common/Services/Emailing/SmtpSender.cs]: No, good catch. I used some images first for testing and I forgot this. I'll update it!

**Review by RaduCStefanescu — DISMISSED** [review]

**Review by RaduCStefanescu — APPROVED** [review]

**Pull request merged**